controllers/TransitionBase.js

```diff
@@ -115,41 +112,38 @@ define(["dcl/dcl", "dojo/when", "dojo/Deferred", "dojo/promise/all", "../Control
 
 			// _displayView is called to show a view, it will handle nested views by calling _displayParents
 			_displayView: function (viewTarget, event, isParent, viewPath) {
-				var deferred = new Deferred();
 				var subEvent;
 				event.dapp.isParent = isParent;
 				event.dapp.viewPath = viewPath;
-				var self = this;
 				// wait for parents to be displayed first
-				when(this._displayParents(viewTarget, event, isParent, viewPath),
-					function (value) {
-						subEvent = Object.create(event);
-						subEvent.dest = viewTarget.split(",").pop();
-						subEvent.dapp.viewPath = viewPath;
-						subEvent.dapp.viewPath.dest = subEvent.dest;
-						subEvent.dapp.isParent = isParent;
+				return Promise.resolve(this._displayParents(viewTarget, event,
+						isParent, viewPath)).then(function (value) {
+					subEvent = Object.create(event);
+					subEvent.dest = viewTarget.split(",").pop();
+					subEvent.dapp.viewPath = viewPath;
+					subEvent.dapp.viewPath.dest = subEvent.dest;
+					subEvent.dapp.isParent = isParent;
 
-						subEvent.dapp.parentView = value.dapp.nextView;
-						var p = self._getParentNode(subEvent) || document.body;
-						if (!self._parentIsValid(p, subEvent.dest, deferred, value)) {
-							return; // p is invalid
-						}
-						subEvent.dapp.parentNode = p;
+					subEvent.dapp.parentView = value.dapp.nextView;
+					var p = this._getParentNode(subEvent) || document.body;
+					if (!this._parentIsValid(p, subEvent.dest, value)) {
+						return; // p is invalid
+					}
+					subEvent.dapp.parentNode = p;
```

Comment:
Something doesn't match here.   The _parentIsValid() method currently takes a `resolve` parameter, yet you aren't passing one.  Of course, I would prefer that it returned a promise rather than taking a resolve parameter, but in any case, it seems like the current code just doesn't work?